src/prime_rl/trainer/config.py

```diff
@@ -148,6 +170,14 @@ def trust_remote_code_only_with_hf(self):
             if self.impl != "hf":
                 raise ValueError("Trust remote code is only supported with the HF implementation.")
         return self
+    
+    @model_validator(mode="after")
+    def random_init_only_with_meta(self):
+        """Random initialize is only supported with the custom implementation."""
+        if self.debug.random_init:
+            if self.impl != "custom":
```

Comment:
oh I think it should work as long as meta load is used. even if the impl is the HF or trust remote code on